django_tasks/backends/database/models.py

```diff
@@ -124,6 +126,30 @@ def task_result(self) -> "TaskResult[T]":
 
         return result
 
+    @property
+    def canonical(self) -> Dict:
+        """
+        Get canonical form
+        """
+        return {
```

Comment:
Suggestion: Because `TaskResult` is a dataclass, you can call `asdict` on it to get it as a dict natively, and then massage the types into something JSON-serializable.

Reply:
I moved `.canonical` inside the base task result, but for task I only wanted to include invariant values. I don't think status, started_at, or finished_at should form part of the signature.